src/stateMachines/activation.test.ts

```diff
@@ -110,15 +113,94 @@ describe('Activation State Machine', () => {
     sendSpy = jest.spyOn(devices[clientId].ClientSocket, 'send').mockReturnValue()
     signStringSpy = jest.spyOn(activation.signatureHelper, 'signString').mockReturnValue('abcdef')
-    // activation.configurator.profileManager = {
-    //   getAmtPassword: jest.fn(),
-    //   getMEBxPassword: jest.fn(),
-    //   getAmtProfile: jest.fn()
-    // } as any
+    currentStateIndex = 0
+    config = {
+      services: {
+        'get-amt-profile': Promise.resolve({
+          clientId: clientId,
+          profile: { profileName: 'ccm', activation: 'ccmactivate', amtPassword: 'Intel123!' }
+        }),
+        'get-amt-domain': Promise.resolve({
+          amtDomain: {
+            profileName: 'vpro',
+            domainSuffix: 'vprodemo.com',
+            provisioningCert: cert,
+            provisioningCertStorageFormat: cert,
+            provisioningCertPassword: 'P@ssw0rd',
+            tenantId: ''
+          }
+        }),
+        'send-generalsettings': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: { AMT_GeneralSettings: { DigestRealm: 'Digest:A3829B3827DE4D33D4449B366831FD01' } }
+          }
+        }),
+        'error-machine': async (_, event) =>
+          await new Promise((resolve, reject) => {
+            setTimeout(() => { resolve({ clientId: event.clientId }) }, 50)
+          }),
+        'send-hostbasedsetup': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: {
+              IPS_HostBasedSetupService: {
+                AllowedControlModes: [2, 1],
+                ConfigurationNonce: 'SkqopmngrtkhdvcteznRbEdgqpc='
+              }
+            }
+          }
+        }),
+        'send-certificate': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: { AddNextCertInChain_OUTPUT: { ReturnValue: 0 } }
+          }
+        }),
+        'send-adminsetup': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: { AdminSetup_OUTPUT: { ReturnValue: 0 } }
+          }
+        }),
+        'send-setup': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: { Setup_OUTPUT: { ReturnValue: 0 } }
+          }
+        }),
+        'send-to-change-amt-password': Promise.resolve({
+          Envelope: {
+            Header: {},
+            Body: { SetAdminAclEntryEx_OUTPUT: { ReturnValue: 0 } }
+          }
+        }),
+        'save-device-secret-provider': Promise.resolve(true),
+        'save-device-mps': Promise.resolve(true),
+        'unconfiguration-machine': Promise.resolve({ clientId }),
+        'network-configuration-machine': Promise.resolve({ clientId }),
+        'features-configuration-machine': Promise.resolve({ clientId }),
+        'cira-machine': Promise.resolve({ clientId }),
+        PROVISIONED: () => {},
+        DELAYED_TRANSITION: () => {}
+      },
+      guards: {
```

Comment:
do we wanna leave these commented out? or remove them entirely

Reply:
done